platform-hub-api/app/services/announcements_processor_service.rb

```diff
@@ -0,0 +1,84 @@
+class AnnouncementsProcessorService
+
+  def initialize email_batch_size, logger
+    @email_batch_size = email_batch_size
+    @logger = logger
+  end
+
+  def run
+    Announcement.published.awaiting_delivery.each(&method(:process))
+  end
+
+  private
+
+  def process announcement
+    @logger.info "Processing announcement #{announcement.id} - #{announcement.level} - '#{announcement.title}'"
+
+    begin
+      announcement.update! status: :delivering
+
+      unless announcement.deliver_to.blank?
+        process_for_email_delivery announcement
+        process_for_slack_delivery announcement
+      end
+
+      # At this point we have to assume that any delivery mechanism triggered has
+      # worked as expected
+      announcement.update! status: :delivered
```

Comment:
As discussed f2f I think this is OK for now but we might need to revisit if we need to ensure that announcements have been really delivered - in case we change / add more distribution mechanisms and are able to get delivery status.